### ISSUE TITLE
Wallet Redesign

### ISSUE THREAD

**Issue by jarradh** [status-im/status-react#584]
the current wallet icon is abit too small, I'm sure we can improve this.
any ideas?

**Comment by rabbit** [comment]
additionally, a wallet is a "top level app" for lack of a better phrase. if we can't reorder contacts I would suggest at least pinning the wallet app to the top of the list.

**Comment by jarradh** [comment]
```
rabbit [5:28 PM] 
heh, maybe wallet app should have a notice at top saying the old cryptocurrency adage “only put on this device what you’re prepared to lose"

[5:29]  
the wording is scary, I’d make it nicer, but I’m a huge fan of that kind of communication with the user

jarradhope [5:29 PM] 
yeah, i think it's very important to encourage only a certain amount of cash, i was thinking of allowing the user to set their 'cash in pocket' limit and have the rest automatically siphoned off into a warm/cold wallet

rabbit [5:29 PM] 
that! is a good idea.

[5:30]  
I would even force that on them. you cant not have a limit. not until 2022

jarradhope [5:30 PM] 
i agree @rabbit and thank-you for thinking about these things, we'll definitely want to make this as safe, secure and trusted for end users

rabbit [5:31 PM] 
don’t mean to dwell on the point. I know you get it.

jarradhope [5:31 PM] 
not dwelling at all, i'm having a blast chatting about it

[5:32]  
yes, sorry I meant there would be a set default, of like, 100eth or some other arbitrary number

[5:32]  
nvm not possible

rabbit [5:32 PM] 
ya but don’t let them disable it to 0 or whatever.

jarradhope [5:32 PM] 
my ninjaedit i mean

rabbit [5:32 PM] 
haha

jarradhope [5:33 PM] 
yeah good point

rabbit [5:33 PM] 
the “I’ll take the risks” user needs to be smacked cause this isn’t the time for that. it’s negligent.

[5:34]  
I come from agency life, these are my babies we’re talking about. :angel:

[5:34]  
anyway good convo back to issues

jarradhope [5:35 PM] 
right, but say with physical wallets the amount of cash does have an upper limit

[5:36]  
because notes and wallet take up space

rabbit [5:36 PM] 
ooh that idea is fun to think of visually. to have a wallet (maybe icon?) that appears more and more full

jarradhope [5:36 PM] 
and are further constrianed by pocket size

[5:36]  
i love that idea @andmironov ^
```

**Comment by jarradh** [comment]
We should think about how to make Status 'feel fat' the closer the Wallet gets to its max limit

**Comment by jarradh** [comment]
@rabbit i think we'll need to have an option like "default wallet", because in Status, we don't want to mandate that our wallet is the only wallet you can use, you should be able to use myetherwallet, for example

**Comment by rabbit** [comment]
best option would be to allow reordering of apps. it's what I tried to do without thinking. defaults tend to become problematic

**Comment by jarradh** [comment]
Reordering of apps like in #585 ?

**Comment by rabbit** [comment]
that screen depicts reordering object type categories, I would want to reorder within a category as well.

**Comment by jarradh** [comment]
Yeah, it's supposed to be there by @andmironov hasn't finished it and is on holidays

**Comment by jarradh** [comment]
Converted this into an Epic and more generally a wallet redesign to support erc20 token and multiple accounts whilst retaining all existing functionality.

**Comment by jeluard** [comment]
# New Wallet progress

## Current state

Development focused at first on simplified main and transactions screens.

## Next steps

Next focus is to bring wallet on-par with existing DApp wallet:

Updated: 05/10/17

- [x]  allow to send transactions (@flexsurfer, done)
- [x]  allow to request transactions from contact (@flexsurfer, done)
- [x]  unsigned transactions should appear in transactions screen (@yenda, done)
- [x]  sending in chat transaction use new wallet UI / data structure (@flexsurfer, done)
- [ ]  old DApp is removed (@flexsurfer, in progress)
- [x]  old transactions UI is removed (@flexsurfer, done)
- [x]  old transactions in drawer is removed (@flexsurfer, done)

## Complete Wallet implementation

Mockups provide an extensive view of Wallet UI functionalities. Wallet squad team progress is tracked by a number of [issues](https://github.com/status-im/status-react/issues?q=is%3Aissue+is%3Aopen+label%3Awallet).

### Main screen (@jeluard)

- [x]  make https://github.com/status-im/status-react/issues/1851
- [ ]  design review https://github.com/status-im/status-react/issues/1946
- [ ]  improve pull-to-refresh visual https://github.com/status-im/status-react/issues/1858
- [ ]  wallet settings

### Transactions history [#1524](https://github.com/status-im/status-react/issues/1524) (@jeluard, @yenda)

- [ ]  fix UI issues (https://github.com/status-im/status-react/issues/1524)
- [ ]  design review  https://github.com/status-im/status-react/issues/1948
- [ ]  implement filtering
- [x]  bind unsigned transactions
- [x]  bind sign/sign all/delete actions
- [x]  bind transaction details
- [x]  bind user 3 words

### Transaction details screen (@yenda)

- [x]  make and bind where used [#1820](https://github.com/status-im/status-react/pull/1820)
- [x]  data-binding [#1775](https://github.com/status-im/status-react/issues/1775)
- [ ] design review https://github.com/status-im/status-react/issues/1979

### Send transactions [#1525](https://github.com/status-im/status-react/issues/1525) (@angusiguess, @flexsurfer)

- [x]  initial send screen [#1773](https://github.com/status-im/status-react/issues/1773)
- [x]  others screen [#1843](https://github.com/status-im/status-react/issues/1843)
- [ ] design review https://github.com/status-im/status-react/issues/1980
- [x]  QR scanning [#1527](https://github.com/status-im/status-react/issues/1527) [#1853](https://github.com/status-im/status-react/issues/1853)
- [ ]  Choose recipient screen design review https://github.com/status-im/status-react/issues/1947

### Request transactions [#1526](https://github.com/status-im/status-react/issues/1526) (@flexsurfer)

- [x]  make  [#1526](https://github.com/status-im/status-react/issues/1526)
- [ ] design review https://github.com/status-im/status-react/issues/1981

### Multiple assets support [#1722](https://github.com/status-im/status-react/issues/1722)

- [ ]  main screen lists wallet assets (total value % is somehow affected by this)
- [ ]  new assets can be added based on user provided details [#1745](https://github.com/status-im/status-react/issues/1745)
- [ ]  manage existing assets
- [ ]  show per asset history [#1528](https://github.com/status-im/status-react/issues/1528)
- [ ]  show per asset value overview

### Multiple accounts support

To be detailed.

### Portfolio value overview [#1529](https://github.com/status-im/status-react/issues/1529)

To be detailed.

### Cross-concerns

- [ ]  generic error handling [#1713](https://github.com/status-im/status-react/issues/1713)
- [ ]  new-toolbar/toolbar2 does not offer proper content element (e.g. title) centering
- [ ]  move to latest toolbar component https://github.com/status-im/status-react/issues/1835
- [ ]  move to latest list component https://github.com/status-im/status-react/issues/1840

## Open issues

- [x]  improve `total value %` https://github.com/status-im/status-react/issues/1848
- [ ]  improve responsivness [#1834](https://github.com/status-im/status-react/issues/1834)

## Potential amelioration points

- [ ]  improve transitions to have smoother experience https://github.com/status-im/status-react/issues/1765 https://github.com/status-im/status-react/issues/1592
- [ ]  persist transactions history filtering
- [ ]  improve data access [#1762](https://github.com/status-im/status-react/issues/1762)

**Comment by annadanchenko** [comment]
Outcome of the test call, current tasks in order of importance for new Wallet:

## Issues to fix
### Must Scope


 
### Nice to have scope
*********** Nice to have scope - take issues below only if list above is done **************


- [ ] **Eric** Scientific notation for amount is shown on Sent Transaction screen if opened from Unsigned tab or from chat #2054 
- [ ] go back to tx list (history and unsigned)  - scroll to the initial point
- [ ] 9. Recipient with request transaction QR - need to implement  **Julien** #2025
- [ ] Wallet - rework send command #2033 **Jan**
- [ ] 12. Waiting for design team to shown time in Tx history
- [ ] Main wallet screen: ETH in Assets list is almost cut (on small screen) #2088
- [ ] 14. If pull down (finger on blue area) **Julien**
- [ ] 5. View transaction details  #2022 (postponed) **Andrey**
- [ ] 18. show transaction fee on send transaction screen  #1971 (postponed) **Andrey**
- [ ]  the font size for the amount field in the tx list may be bigger
- [ ] the password popup - iOS looks on Android 
- [ ]  Request tx flow - amount field should be fixed when in the scroll state (smaller screen)
- [ ] shadow line - there’s it on the mockup on Android only, desired to implement on Android
- [ ] Tap on Send button highlights extra area when the title is multi-row
- [ ] Wallet name title not aligned on Android, needs to be 72px from the left
- [ ]  wallet selection screen - iOS on Android
- [ ] Amount validation considers only balance and does not consider fee on top of it. It should take into account balance+fee to check that amount to be sent is sufficient. #2009



## Done Issues
### Must scope
- [x]  test **Julien** Wallet -> send: Can't use address from clipboard: Error 'invalid address' #2156
- [x] Second transaction can't be sent if previous transaction is in the Unsigned list #2180
- [x]  **Eric**  in Unsigned - after tx has signed and sent - display the Transaction Sent screen. In send command if tap on Sign Later then "popups" to inform user that its's moved to Unsigned are shown.
- [x] test **Andrey**  fix line that crops content on send and request transaction screens? (screenshot from iphone 6)https://github.com/status-im/status-react/issues/2182
- [x] test **Andrey** borders for button Send and Request (see slack)https://github.com/status-im/status-react/issues/2182
- [x] test **Jan** add timestamp to tx list and need design update for the case with the User name in the tx From field #2131
- [x] test **Eric** Unsigned transactions from 1 account are shown for another account on the same device #2132
- [x] test **Julien** extend tap area for Sign Later and Sign Transactions (Wallet and modal Sign Transaction screens) #2152 
- [x] **Eric** No feedback on insufficient funds if send transaction from Unsigned #2128
- [x] **Andrey**  Error handling on Send Transaction screen if something goes wrong after providing password and tap on Sign Transaction. E.g. try to send all funds - but fee is not taking into account -> need to show some screen/popup? with feedback to the user. #2142
- [x] test **Julien**  QR code amount handling https://github.com/status-im/status-react/issues/2133
- [x] **Eric**  Show warning when app is syncing to tell user that it's not possible to use Wallet (yellow instead of red) #2144 
- [x] **Jan** - If send command in chat: Close Send Transaction screen -> Transaction is removed from Unsigned. 
- [x] **Jan** - Transaction is sent right after tapping on send button (no Send Transaction screen shown) #2031  
- [x] **Julien** - allow `0` as valid amount -> caused regression issues with amount validation. See amount validation regression #2107
- [x]  **Denis** Wallet: Can't scan white QR codes using wallet QR scanner bug high-priority wallet
#2023 - mockups updated  **Julien** can implement it
- [x] App crashed if not allowed to access camera and open Choose Recipient screen (iOS only) #2057 
- [x] 1. Unsigned transactions -> can’t Sign all - error message. We will not implement Sign All. We will implement Sign for each transaction (button Sign under transaction) #1994 **Andrey**
- [x] 2. Error handling for amount - trying to send too much or error in web3 - nothing happens but error should be shown #1990 **Julien** 
- [x] 3. Validation for scanned code/ copy pasted - in choose recipient -> show popup(alert) on same Choose Recipient #1991  **Julien**
- [x] 4. Show loading indicator when Sign Transaction is tapped (same loading as for Sign in screen): #1992 **Angus**
- [x] 6. Syncing line (blue one) is to be removed - **Eric** PR #2011
- [x] 8. Profile (drawer) - remove Unsigned transaction section (to sign you need to go to Wallet) #1994 **Andrey**
- [x] 19. transactions order in history is reversed in sections **Eric** PR #2012
- [x] 15. Update unsigned transactions count when transaction expires and fix count color **Eric** Bug #2003
- [x] **Julien** - 7. Main wallet - 0 usd  
- [x] **Julien**  - Exchange button - need to be 1 line 
- [x] **Angus** - No loading indicator is shown if invoke Sign Transaction from chat or from Unsigned tab #2046  
- [x] **Eric** - #2020 can't scroll transactions to the very end in History  after switching from Unsigned  tab
- [x] **Eric** - 11. Hash for unsigned - add not applicable .. 
- [x] **Andrey** - Labels on Send/Request Transaction screens are cut in the bottom (Android only) #2059 
- [x] **Julien** - Add delete confirmation on Unsigned tab - when user taps on Delete under transaction 
- [x] **Andrey** Wallet balance text is shown behind the screen (takes more space than screen has) #2087
- [x]  test **Andrey**  how do we make the user aware about the network? #2096 
- [x] test **Julien** - iOS Unsigned tab, button `Sign` corner radius is to be added, now looks like square 
- [x] test **Andrey** - 20. remove old wallet dapp #2021 

### Nice to have scope

- [x] **Eric issue #2016** Scientific notation is shown for small amounts in transaction history and unsigned transaction list
- [x] 16. #2018 When going back from details in unsigned transaction the tab should be unsigned transaction not history **Eric** 
- [x] 10. In Unsigned/History - show Recipient name if available both in the lists and in transaction details screen #2040  **Eric**
- [x] extend the tap area of History/Unsigned tabs **Eric**
